de.variantsync.core/src/main/java/de/variantsync/core/ast/JsonParserASTwithLineGrammar.java

```diff
@@ -0,0 +1,73 @@
+package de.variantsync.core.ast;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import de.variantsync.core.interfaces.Grammar;
+
+/**
+ * 
+ * Uses gson to import and export ASTs to and from json. A generic use of grammar is not possible, hence for each grammar a different class has to be created.
+ * In this case the used grammar is LineGrammar.
+ * 
+ * @author jerem
+ *
+ */
+public class JsonParserASTwithLineGrammar {
+
+	static Gson gson = new GsonBuilder().setPrettyPrinting().excludeFieldsWithoutExposeAnnotation().create();
+
+	public static <A extends Grammar, B> String exportAST(AST<A, B> ast) {
+
+		// generic use of A not possible, instead declare used Grammar here
+		Type type = new TypeToken<AST<LineGrammar, B>>() {}.getType();
+
+		return gson.toJson(ast, type);
+	}
+
+	public static <A extends Grammar, B> AST<A, B> importAST(String json) {
+
+		// generic use of A not possible, instead declare used Grammar here
+		Type type = new TypeToken<AST<LineGrammar, B>>() {}.getType();
+
+		return gson.fromJson(json, type);
+	}
+
+	public static <A extends Grammar, B> String exportToFile(Path path, AST<A, B> ast) {
+
+		// generic use of A not possible, instead declare used Grammar here
+		Type type = new TypeToken<AST<LineGrammar, B>>() {}.getType();
+
+		String content = gson.toJson(ast, type);
+		try {
+			Files.writeString(path, content);
+		} catch (IOException e) {
+			e.printStackTrace();
+			return null;
+		}
+
+		return content;
+	}
+
+	public static <A extends Grammar, B> AST<A, B> importFromFile(Path path) {
```

Comment:
fix `A` to LineGrammar if you always return an AST with LineGrammar. (as mentioned above)

Reply:
I also fixed B to String, or should i leave B generic?

Reply:
That depends on whether GSON could also parse and serialise other types than just `String` from Json. If GSON always returns a `String` and we have to manually convert it to `B`, then yes fixing to String makes sense. If however, GSON can directly parse the json contents to an object of type `B`, then we we can stick with `B`.